system/lib/wasmfs/syscalls.cpp

```diff
@@ -889,4 +889,53 @@ long __syscall_rename(long old_path, long new_path) {
   return 0;
 }
 
+long __syscall_symlink(char* old_path, char* new_path) {
+  auto pathParts = splitPath(new_path);
+
+  long err;
+  auto parsedPath = getParsedPath(pathParts, err);
+
+  if (!parsedPath.parent) {
+    return err;
+  }
+
+  if (pathParts.back().size() > WASMFS_NAME_MAX) {
+    return -ENAMETOOLONG;
+  }
```

Comment:
This just happens when we create a new path, I believe. So significantly fewer locations than in general. Still, perhaps we can refactor to avoid even that duplication, but maybe later?